test/interval_tests/consistency.jl

```diff
@@ -388,4 +388,20 @@ setprecision(Interval, Float64)
 
     end
 
+    @testset "Type stability" begin
+        for T in (Float32, Float64, BigFloat)
+
+            x = Interval{T}(3, 4)
+            y = Interval{T}(5, 6)
+
+            for op in (+, -, *, /, atan)
+                @inferred op(a, b)
+            end
+
+            for op in (sin, cos, exp, log, tan, abs, mid, diam)
+                @inferred op(a)
```

Comment:
```suggestion
                @inferred op(x)
```

You probably want to use the intervals you define above.